django-testbed/django-root/xenopus_frog/views.py

```diff
@@ -7,14 +7,57 @@
 from xenopus_frog.models import User
 
 
+class BunniesSerializer(ModelSerializer):
+    """
+    Regulates fields exposed by default & as requested.
+
+    Construct URLs like /?include_fields=first_name&include_fields=email to opt-in into fields.
+
+    By default, all "opt_in_only_fields" are excluded from serializer, unless specifically opted-in.
+
+    If any of fields specified to be included does not belong to opt_in_only_fields, the serializer
+    will perform in "strict" mode that ONLY fields specced are included.
+    """
+
+    def __init__(self, *args, **kwargs):
+        result = super().__init__(*args, **kwargs)
+
+        # only if context.request is available (ie. not during inspection etc)
+        if hasattr(self, "context") and "request" in self.context:
+            heavy_fields = set(getattr(self.Meta, "opt_in_only_fields", []))
+
+            if "include_fields" in self.context["request"].GET:
+                include_fields = dict(self.context["request"].GET)["include_fields"]
+                if not type(include_fields) == list:
```

Comment:
I'm guessing the type here is out of our control so should be safe to assume is only ever a list.. but to be paranoid probably `if not isinstance(include_fields, (list, tuple))` is better

Reply:
as we're doing the splitting (comma separated) this no longer matters - removing type casting